src/test/java/com/example/bookstorewebapp/service/book/BookServiceImplTest.java

```diff
@@ -0,0 +1,168 @@
+
+    @Test
+    @DisplayName("Find Book by ID returns book")
+    void findBookById_ReturnsBook() {
+        Book book = new Book();
+        book.setId(bookId);
+        BookDto responseDto = new BookDto();
+
+        when(bookRepository.findById(bookId)).thenReturn(Optional.of(book));
+        when(bookMapper.toDto(book)).thenReturn(responseDto);
+
+        BookDto actual = bookService.findById(bookId);
+
+        verify(bookRepository, times(1)).findById(bookId);
+        assertEquals(actual, responseDto);
+    }
+
+    @Test
+    @DisplayName("Find Book by ID throws EntityNotFoundException when book does not exist")
+    void findBookById_ThrowsEntityNotFoundException_WhenBookNotExist() {
+        when(bookRepository.findById(bookId)).thenReturn(Optional.empty());
+
+        assertThrows(EntityNotFoundException.class, () -> {
+            bookService.findById(bookId);
+        });
+    }
+
+    @Test
+    @DisplayName("Delete Book by ID successfully")
+    void deleteBookById_Success() {
+        when(bookRepository.existsById(bookId)).thenReturn(true);
+
+        bookService.deleteById(bookId);
+
+        verify(bookRepository, times(1)).deleteById(bookId);
+    }
+
+    @Test
+    @DisplayName("Delete Book by ID throws EntityNotFoundException when book does not exist")
+    void deleteBookById_ThrowsEntityNotFoundException_WhenBookNotExist() {
+        when(bookRepository.existsById(bookId)).thenReturn(false);
+
+        assertThrows(EntityNotFoundException.class, () -> {
+            bookService.deleteById(bookId);
+        });
+    }
+
+    @Test
+    @DisplayName("Find all Books returns books")
+    void findAllBooks_ReturnsBooks() {
+        Pageable pageable = Pageable.unpaged();
+        Book book = new Book();
+        BookDto responseDto = new BookDto();
+
+        when(bookRepository.findAll(pageable)).thenReturn(new PageImpl<>(List.of(book)));
+        when(bookMapper.toDto(book)).thenReturn(responseDto);
+
+        List<BookDto> actual = bookService.findAll(pageable);
+
+        verify(bookRepository, times(1)).findAll(pageable);
+        assertEquals(List.of(responseDto), actual);
+    }
+
+    @Test
+    @DisplayName("Search Books returns books")
+    void searchBooks_ReturnsBooks() {
+        Pageable pageable = Pageable.unpaged();
+        BookSearchParameters params = new BookSearchParameters(
+                new String[]{"Title1"}, new String[]{"Author1"});
+        Book book = new Book();
+        BookDto responseDto = new BookDto();
+
+        when(specificationBuilder.build(params)).thenReturn((Specification<Book>)
+                (root, query, criteriaBuilder) -> criteriaBuilder.conjunction());
+        when(bookRepository.findAll(any(Specification.class), any(Pageable.class)))
+                .thenReturn(new PageImpl<>(List.of(book)));
+        when(bookMapper.toDto(book)).thenReturn(responseDto);
+
+        List<BookDto> actual = bookService.search(params, pageable);
+
+        verify(bookRepository, times(1))
```

Comment:
```suggestion
        verify(bookRepository)
```
times 1 is default value